### PR TITLE
Plugin Catalog: support Grafana instances that cannot communicate with gcom

### LINKED ISSUE #39459: Plugins Catalog: support Grafana instances that cannot connect to GCOM
The Plugins Catalog needs to support an instance of Grafana that isn't able to contact the GCOM api. To achieve that we should do the following:

- [ ] Make the store aware if GCOM is inaccessible
- [ ] Render the Plugins Catalog list with limited features / functionality if GCOM is inaccessible (e.g filter by ALL)
- [ ] Navigate to Plugin Details page with limited features / functionality if GCOM is inaccessible (e.g Install Controls)
- [ ] Render a notice (somewhere) that explains that because GCOM isn't accessible the catalog has limited functionality

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
- [x] Have a selector for checking if GCOM is not available
- [x] Disable filtering by Installed / All on the Plugins Catalog list page if GCOM is inaccessible 
- [x] Disable install controls on the Plugin Details page if GCOM is inaccessible
- [ ] Render a notice (somewhere) that explains that because GCOM isn't accessible the catalog has limited functionality
- [x] Add tests

**Which issue(s) this PR fixes**:
Fixes #39459

---

**The browse with the limited filtering when GCOM is not available:**
Screenshot 2021-09-28 at 14 39 06
 

**The plugin details with disabled install controls if GCOM is not available:**
Screenshot 2021-09-28 at 14 39 20

### CODE REVIEW

File: public/app/features/plugins/admin/pages/PluginDetails.test.tsx
Comment (jackw) [inline public/app/features/plugins/admin/pages/PluginDetails.test.tsx:378]: If you're gonna go with the earlier suggestion this will need updating too.

```suggestion
    const message = 'The functionality to install and uninstall plugins has been disabled because the Grafana server cannot access grafana.com.';
```

**Review by jackw — COMMENTED** [review]
Lovely additions @leventebalogh! 👏 

I left a minor content change. As I opened this PR I cannot approve it "officially" but it LGTM! 🚀

**Review by leventebalogh — APPROVED** [review]
Approving after reviewing it together with @jackw.

**Pull request merged**
